doop.c

```diff
@@ -657,17 +688,72 @@ Magic and tainting are handled.
 =cut
 */
 
+static SV *
+S_concat_maybe_overloaded(pTHX_ SV *lhs, SV *rhs)
+{
+    STRLEN len;
+    const char *s;
+
+    /* We need both ROK and RMG in order to even have a chance of having
+     * concat overload, so we might be able to speed this up with: */
+    /* U32 flags = SvFLAGS(lhs)|SvFLAGS(rhs); */
+    /* if (UNLIKELY((flags & (SVf_ROK|SVs_GMG)))) { */
+    if (UNLIKELY(S_sv_has_magic_method(aTHX_ lhs, concat_amg)
+              || S_sv_has_magic_method(aTHX_ rhs, concat_amg))) {
```

Comment:
once these have been added to embed.fnc the S_ prefix should removed, along with (aTHX_ as well. Eg, by putting this function in embed.fnc you get the appropriate macros which manage this for you. It also makes life easier if the function is changed from static to exported as no code needs to be changed.